### PR TITLE
PUB-569 search validation

### PR DESCRIPTION
**Pull request by commitBlob**
### JIRA link ###
https://tools.hmcts.net/jira/browse/PUB-569

### Change description ###
- prevent form resubmit on refresh, back and forward browser clicks
- set autocomplete to overlay to avoid focus issues
- link errors in the error summary to the answer they relate to

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/views/search.njk [hunk 1]
```diff
@@ -20,7 +20,7 @@
                     errorList: [
                         {
                             text: errorMessage,
-                            href: "#"
+                            href: "#search-input-container"
```
Comment (kurt-uk) [inline src/main/views/search.njk:23]: the anchor tag here doesnt return focus to the input from when i tried it, not sure if youre using the right ref, also if you are using the right ref, can we rename? having `.../#search-input-container` in the url doesnt look fab

File: src/main/views/search.njk [hunk 2]
```diff
@@ -70,12 +70,21 @@
...
           element: element,
           id: 'search-input',
-          source: source
-        })
+          source: source,
+          displayMenu: 'overlay'
+        });
+      history.pushState(null, null, location.href);
+      history.back();
+      history.forward();
```
Comment (kurt-uk) [inline src/main/views/search.njk:83]: this seems extremely hacky? forcing them back and forward?

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/views/search.njk (hunk 2 above)
Comment (commitBlob) [reply src/main/views/search.njk:83]: This was the only solution I found that covered back, forward, and refresh cases.
Others only worked for back and refresh

File: src/main/views/search.njk (hunk 1 above)
Comment (commitBlob) [reply src/main/views/search.njk:23]: It doesn't focus it jumps to that element on the page.

File: src/main/views/search.njk [hunk 3]
```diff
@@ -45,7 +45,7 @@
                 {% if noResultsError or invalidInputError %}
                     <span class="govuk-error-message">Enter a court name, town or city</span>
                 {% endif %}
-                <div class="{{ inputClasses }}" id="search-input-container"></div>
+                <div class="{{ inputClasses }}" id="search-input"></div>
```
Comment (kurt-uk) [inline src/main/views/search.njk:48]: we have 2 elements with the same id now, youll have to either change the name of this again or change the id in the autcomplete

**Comment by ChrisS1512** [comment]
I'm getting an issue where the browsers back button isn't working in the following scenario:

1) Navigate to page and search for something that causes an error
2) Click the link in the error text
3) Click the browsers back button, which doesn't then take you back

**Review by ChrisS1512 — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]

**Pull request merged**